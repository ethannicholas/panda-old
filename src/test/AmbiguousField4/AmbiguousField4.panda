================================================================================
The reference to "t.length:=" is ambiguous because the class has both a field 
and an assignment method by that name. Outside of the class with both field and
method visible, we should complain about ambiguity.
================================================================================
@final
class AmbiguousField4 {
    var length:Int := 5

    @self
    method length:=(i:Int) {
        Console.writeLine("calling length:=")
    }
}

method main() {
    var t := AmbiguousField4()
    t.length := 0
    Console.writeLine("shouldn't see this")
}