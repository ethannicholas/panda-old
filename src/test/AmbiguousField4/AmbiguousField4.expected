error: AmbiguousField4.panda:18:6: reference to 'length' is ambiguous, both field length (of type panda.core.Int32) and method length:=() match
1 error
